### DISCUSSION TITLE
Custom component functions

### DISCUSSION THREAD

**Discussion by rootiest** [michaelbrusegard/tabline.wez#34] [Q&A] [answered]
Hi, I'm trying to make a custom tab index component that will add the "ordinal suffix" ie:
1st
2nd
3rd
etc..

This should be fairly straightforward to do, I did it with my bufferline in neovim.
I have no problem writing the function, but I'm having some difficulty with the correct syntax to get/use the TabInformation object so I can retrieve the tab index number.

The "Hello" example in the readme doesn't show using the Window or TabInformation objects though the documentation mentions they can get passed to your custom functions.

Can anyone show any example of how this works to help me out?

Thanks!

**Comment by rootiest** [comment, marked as answer]
Scratch that, I figured out why it wasn't working.

I overlooked that `tab_index` was changed to `index` very recently and was trying to use the incorrect component name for my tabline version.

Here is my code if anyone wants to do the same:

```lua
-- Function to convert a number to its ordinal string representation
local function ordinal(n)
	local suffixes = { "th", "st", "nd", "rd" }
	local mod100 = n % 100
	return n .. (suffixes[(mod100 - 20) % 10 + 1] or suffixes[1])
end
```
Then this component:
```lua
{
	"index",
	fmt = function(n)
		return ordinal(n)
	end,
	padding = { left = 0, right = 0 },
},
```

**Comment by michaelbrusegard** [comment]
Yea, sorry for that change last night. I am not that good at how to announce breaking changes since I have not maintained something like this before. I tried to create a new release with the changes. I am glad you got it working

**Reply by rootiest** [reply]
All good! No worries!

It reminded me that I should write a lua function to periodically update my plugins haha
